src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -90,6 +90,32 @@ internal sealed class VariableInformation : IVariableInformation
         private string DisplayHint { get; set; }
         public bool IsPreformatted { get; set; }
 
+        private static readonly string[] s_validAddressFormats = new string[] {
+            @"^0x[0-9a-fA-F]+$",
+            @"^(0x[0-9a-fA-F]+)\b"
+        };
+
+        public string Address()
+        {
+            // ask GDB to evaluate "&expression"
+            string command = "&("+FullName()+")";
+            var result = EvalDependentExpression(command);
+            for (int i = 0; i < s_validAddressFormats.Length; i++)
+            {
+                if (Regex.IsMatch(result, s_validAddressFormats[i]))
+                {
+                    return result;
+                }
+            }
+            return null;
```

Comment:
I believe you want to throw an exception here. Something with a message like: $"Unexpected result '${result}' when evaluating '${command}".